spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/launcher/TaskLauncherSink.java

```diff
@@ -60,7 +65,10 @@ public void taskLauncherSink(TaskLaunchRequest request) {
 	private void launchTask(TaskLaunchRequest taskLaunchRequest) {
 		Assert.notNull(taskLauncher, "TaskLauncher has not been initialized");
 		logger.info("Launching Task for the following resource " + taskLaunchRequest);
-		Resource resource = delegatingResourceLoader.getResource(taskLaunchRequest.getUri());
```

Comment:
actually I think you would still want to use the resource loader here... you just need to make sure that the `MavenResourceLoader` that is configured within the `DelegatingResourceLoader` has `remoteRepositories` provided on its `MavenProperties`